bfvmm/include/memory_manager/page.h

```diff
@@ -0,0 +1,136 @@
+//
+// Bareflank Hypervisor
+//
+// Copyright (C) 2015 Assured Information Security, Inc.
+// Author: Rian Quinn        <quinnr@ainfosec.com>
+// Author: Brendan Kerrigan  <kerriganb@ainfosec.com>
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+
+#ifndef PAGE_H
+#define PAGE_H
+
+#include <stdint.h>
+#include <memory.h>
+
+class page
+{
+public:
+
+    /// Page Default Constructor
+    ///
+    /// Creates an empty, invalid page
+    ///
+    page();
+
+    /// Valid Page Constructor
+    ///
+    /// If given the correct values, creates a valid page
+    ///
+    /// @param phys the physical address of the page
+    /// @param virt the virtual address of the page
+    /// @param size the size of the page in bytes
```

Comment:
Probably shouldn't allow size in bytes to be specified, in the base class at least. Support for large and huge pages should probably have their own class, and the size be fixed for each class.